libs/common/Parsing/RespParsingException.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Garnet.common.Parsing
+{
+    /// <summary>
+    /// Exception wrapper for RESP parsing errors.
+    /// </summary>
+    public class RespParsingException : GarnetException
+    {
+        /// <summary>
+        /// Construct a new RESP parsing exception with the given message.
+        /// </summary>
+        /// <param name="message">Message that described the exception that has occurred.</param>
+        RespParsingException(String message) : base(message)
+        {
+            // Nothing...
+        }
+
+        /// <summary>
+        /// Throw an "Unexcepted Token" exception.
+        /// </summary>
+        /// <param name="c">The character that was unexpected.</param>
+        [DoesNotReturn]
+        public static void ThrowUnexpectedToken(byte c)
+        {
+            Throw($"Unexpected byte ({c}) in RESP command package.");
```

Comment:
since tokens are ASCII, I wonder if it would be preferable to use `{(char)c}` here, or maybe `{(char)c} ({c})` if we're worried about random corruption

Reply:
I think this would make the error messages more readable. We just need to be careful to escape control characters.

Reply:
This has been addressed with the last commits.